WikiWatershed/model-my-watershed
# Collections API: Update Celery to be Synchronous

## Overview

Update Celery to use the new synchronous geoprocessing service, instead of the older Spark JobServer setup. See original issue and commit messages for details.

Connects #2102 

## Testing Instructions

 * Ensure your Worker VM has been provisioned at least once on this `feature/collections-api` branch
 * ~~Get the latest version of [mmw-geoprocessing](https://github.com/WikiWatershed/mmw-geoprocessing) and run [`cibuild`](https://github.com/WikiWatershed/mmw-geoprocessing/blob/develop/scripts/cibuild) to generate `api-assembly-3.0.0-alpha.jar`~~ 
 * Check out this branch
 * ~~Copy that new JAR into `/opt/geoprocessing/mmw-geoprocessing-3.0.0-alpha.jar` in the Worker VM, and then reload the Worker VM~~
 * Provision the Worker
 * Open the app [:8000/](http://localhost:8000/)
 * Draw any shape, and proceed to Analyze. Ensure that all analyses complete successfully.
 * Proceed to TR-55. Draw modifications, move the precipitation slider. Ensure that all model runs complete successfully.
